framework/include/utils/ExtraIDUtils.h

```diff
@@ -0,0 +1,29 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "MooseTypes.h"
+#include "libmesh/elem.h"
+#include "libmesh/mesh_base.h"
+#include "libmesh/dof_object.h"
+
+namespace ExtraIDUtils
+{
+/**
+ * create a new set of extra id by parsing the existing extra ids
+ * @param mesh input mesh
+ * @param block_ids block ids
+ * @param extra_ids extra ids
+ * @return map of element id to new extra id
+ **/
+std::map<dof_id_type, dof_id_type> ParseExtraIDs(const MeshBase & mesh,
```

Comment:
- I don't think the name for this method properly describes what it does, and the docstring is definitely lacking. 
- Method names should start with a lower cased character
- This should probably go into MooseMeshUtils

Reply:
Renamed as `getExtraIDUniqueCombinationMap`